BackEnd/routes/program.js

```diff
@@ -190,32 +189,68 @@ router.post(
 router.delete(
   "/delete-program",
   [verifyAccessToken, getPermissions],
-  (req, res) => {
-    if (req.permissions.userRole != "owner") {
+  async (req, res) => {
+    if (req.permissions.userRole !== "owner") {
       return res.status(403).json({ error: "Forbidden" });
-    } else {
-      const programId = req.query.programID;
+    }
+
+    const programId = req.query.programID;
+
+    if (!programId) {
+      return res.status(400).json({ error: "Program ID is required" });
+    }
+
+    const session = await mongoose.startSession();
+    session.startTransaction();
+
+    try {
+      const program = await Programs.findById(programId).session(session);
 
-      if (!programId) {
-        return res.status(400).json({ error: "Program ID is required" });
+      if (!program) {
+        await session.abortTransaction();
+        session.endSession();
+        return res.status(404).json({ error: "Program not found" });
       }
-      console.log("ProgramID:",programId);
-      Programs.findByIdAndDelete(programId)
-        .then((deletedProgram) => {
-          if (!deletedProgram) {
-            return res.status(404).json({ error: "Program not found" });
-          }
-          console.log("Successfully deleted program");
-          return res.status(200).json({});
-        })
-        .catch((error) => {
-          console.error(error);
-          return res.status(500).json({ error: "Internal Server Error" });
-        });
+
+      const ownerId = program.owner;
+      const editors = program.editors;
+      const viewers = program.viewers;
+
+      await Programs.findByIdAndDelete(programId).session(session);
+
+      await Users.updateOne(
+        { _id: owner },
+        { $pull: { programs: { _id: programId } } }
+      ).session(session);
+
+
+      await Users.updateMany(
```

Comment:
Currently it will just remove the program from the User's list of programs. In the future we should implement a way to notify people that it was deleted by the owner otherwise they just won't find the program anymore and won't know what happened to it.